### PR TITLE
Simplify, and fix ingestion --overwrite

### PR DESCRIPTION
**Pull request by soumitradev**

### CODE REVIEW

**Review by majimearun — CHANGES_REQUESTED** [review]
works for all cases except for when timetables are already created for the old ingested data.

script will need to delete those timetables (only pertaining to the old data of the same semester and not archived ones of others)

**Comment by soumitradev** [comment]
Will close #98

File: backend/src/ingestJSON.ts [hunk 1]
```diff
@@ -95,331 +98,150 @@ export const ingestJSON = async (
...
+            .where("acad_year = :year", { year: latestCourse.acadYear })
+            .andWhere("semester = :semester", {
+              semester: latestCourse.semester,
+            })
+            .execute();
+          console.log(
+            `deleted ${deletedCourseResult.affected} courses from academic year ${latestCourse.acadYear} and semester ${latestCourse.semester}!`,
+          );
+        }
+      }
```
Comment (majimearun) [inline backend/src/ingestJSON.ts:146]: sections of the course arent being deleted

**Review by majimearun — CHANGES_REQUESTED** [review]
except that one missing delete LGTM

File: backend/src/ingestJSON.ts (hunk 1 above)
Comment (soumitradev) [reply backend/src/ingestJSON.ts:146]: sections are configured to delete when their parent course is deleted anyways, so this isn't an issue

Comment (soumitradev) [reply backend/src/ingestJSON.ts:146]: see: https://github.com/crux-bphc/chronofactorem-rewrite/blob/fix-ingestion/backend/src/entity/Section.ts#L27

Comment (majimearun) [reply backend/src/ingestJSON.ts:146]: cool

**Review by majimearun — APPROVED** [review]
LGTM

**Pull request merged**